spec/services/workflows/non_passported_workflow_spec.rb

```diff
@@ -2,12 +2,48 @@
 
 module Workflows
   RSpec.describe NonPassportedWorkflow do
-    let(:assessment) { create :assessment }
+    let(:assessment) { create :assessment, applicant: applicant }
+
     describe '.call' do
-      it 'raises' do
-        expect {
-          described_class.call(assessment)
-        }.to raise_error RuntimeError, 'Not yet implemented: Check Fincancial Eligibility service currently does not handle non-passported applicant'
+      subject { described_class.call(assessment) }
+
+      context 'self_employed' do
+        let(:applicant) { create :applicant, self_employed: true }
+
+        it 'calls the self-employed workflow' do
+          expect(SelfEmployedWorkflow).to receive(:call).with(assessment)
+          subject
+        end
+      end
+
+      context 'not employed, not self_employed, Gross income exceeds threshold' do
```

Comment:
Should not employed be a mock value from a factory method? As it's not explicit that the person is unemployed.